### PR TITLE
fix: category suggestions

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type** 🐞 `Fix`

**Description** 
1. Fixes handling of category suggestions for 
   - [x] React [🎥  Loom](https://www.loom.com/share/f83b644b057b42ce9b6d002d68942d52)
   - [x] Vue [🎥  Loom](https://www.loom.com/share/ac8e77e794dc48c284d9acc04dc20951)
   - [x] React-Native [🎥  Loom](https://www.loom.com/share/bf25bdee2009460b9aa914eba2a73301)

Docs 📜  https://github.com/appbaseio/Docs/pull/303

### CODE REVIEW

File: packages/searchbase/src/SearchComponent.js [hunk 1]
```diff
@@ -1030,6 +1032,9 @@ class SearchComponent extends Base {
     prevValue: any,
     nextValue: any
   ): void {
+    if (options.category) {
```
Comment (bietkul) [inline packages/searchbase/src/SearchComponent.js]: @mohdashraf010897 Why did you add `category` in options?

**Review by bietkul — CHANGES_REQUESTED** [review]
@mohdashraf010897 Explain the `SearchComponent` changes? It seems that you've added a new option to `setter` methods.

File: packages/searchbase/src/SearchComponent.js (hunk 1 above)
Comment (mohdashraf010897) [reply packages/searchbase/src/SearchComponent.js]: @bietkul 

https://www.loom.com/share/0a1ec3558a5e42edade89c4fa6f9b886

**Review by bietkul — CHANGES_REQUESTED** [review]
@mohdashraf010897 
- Define `setCategoryValue` method in `SearchComponent` class. Implementation should be similar to other setter methods like `setValue`.
- Remove/Update unnecessary code where you're using `options.category`.
- SearchBox: First set `category` using `setCategoryValue` method. It should be a silent update (`triggerDefaultQuery` and `triggerCustomQuery` should be `false`) then update `value` using `setValue` and trigger queries.
- Update docs for `setCategoryMethod`.

**Comment by mohdashraf010897** [comment]
> @mohdashraf010897
> 
> * Define `setCategoryValue` method in `SearchComponent` class. Implementation should be similar to other setter methods like `setValue`.
> * Remove/Update unnecessary code where you're using `options.category`.
> * SearchBox: First set `category` using `setCategoryValue` method. It should be a silent update (`triggerDefaultQuery` and `triggerCustomQuery` should be `false`) then update `value` using `setValue` and trigger queries.
> * Update docs for `setCategoryMethod`.

@bietkul 
https://www.loom.com/share/28c3f47a1d0f48b490cbe5695f336e78

File: packages/searchbase/src/SearchComponent.js [hunk 2]
```diff
@@ -562,7 +562,9 @@ class SearchComponent extends Base {
       customQuery: this.customQuery ? this.customQuery(this) : undefined,
       defaultQuery: this.defaultQuery ? this.defaultQuery(this) : undefined,
       ...(this.value && { value: this.value }),
-      categoryValue: this.categoryValue,
+      ...(this.type !== queryTypes.Suggestion
```
Comment (bietkul) [inline packages/searchbase/src/SearchComponent.js]: Category value should apply irrespective of the query type.

File: packages/searchbase/src/SearchComponent.js [hunk 3]
```diff
@@ -700,6 +702,13 @@ class SearchComponent extends Base {
     }
   };
 
+  // Method to set the category value
+  setCategoryValue = (value: any, options?: Options = defaultOptions): void => {
+    const prev = this.categoryValue;
+    this.categoryValue = value;
+    this._applyOptions(options, 'categoryValue', prev, this.value);
```
Comment (bietkul) [inline packages/searchbase/src/SearchComponent.js:707]: @mohdashraf010897 It seems you've missed handling `categoryValue` in apply options method.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/searchbase/src/SearchComponent.js (hunk 2 above)
Comment (mohdashraf010897) [reply packages/searchbase/src/SearchComponent.js]: But we don't apply categoryValue for type `suggestion` query in reactivesearch.

File: packages/searchbase/src/SearchComponent.js (hunk 3 above)
Comment (mohdashraf010897) [reply packages/searchbase/src/SearchComponent.js:707]: How? I think it's dynamically handled 
image

Comment (bietkul) [reply packages/searchbase/src/SearchComponent.js:707]: ok

File: packages/searchbase/src/SearchComponent.js (hunk 2 above)
Comment (bietkul) [reply packages/searchbase/src/SearchComponent.js]: It is not related to `reactivesearch`, if `categoryValue` is present or passed by user then it should be applicable. Backend doesn't have any restrictions.

File: packages/react-searchbox/src/components/SearchComponent.js [hunk 4]
```diff
@@ -100,12 +100,16 @@ class SearchComponent extends React.Component {
       maxPopularSuggestions,
       stopwords
     } = this.props;
-    let { value } = this.props;
+    let { value, categoryValue: category } = this.props;
     if (window && window.location && window.location.search) {
```
Comment (bietkul) [inline packages/react-searchbox/src/components/SearchComponent.js:104]: @mohdashraf010897 This is not the right place to handle URL params. Check the implementation in `URLParamsProvider` and remove it from `SearchComponent`.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/react-searchbox/src/components/SearchComponent.js (hunk 4 above)
Comment (mohdashraf010897) [reply packages/react-searchbox/src/components/SearchComponent.js:104]: isn't this an existing thing?

Comment (bietkul) [reply packages/react-searchbox/src/components/SearchComponent.js:104]: Yes, added by you.
